daceml/autodiff/library/library.py

```diff
@@ -0,0 +1,351 @@
+"""
+Dace library for autodiff
+
+Includes the BackwardPass library node, and the replacements for the python frontend
+"""
+from typing import Dict, Set, Optional
+import copy
+
+import dace
+import dace.library
+from dace import SDFG, SDFGState, nodes, data, properties
+from dace.transformation import transformation as pm
+from dace.transformation.passes import analysis
+from dace.sdfg import graph
+
+from daceml.autodiff import backward_pass_generator as engine, analysis as autodiff_analysis
+from daceml.util.utils import in_edge_with_name
+
+
+@properties.make_properties
+class Array(data.Array):
```

Comment:
In dace's `replacements.py`, you can see that the binop replacements match on the operand's `__class__.__name__`. This means that naming this anything else prevents the replacements from matching, and then parameters can't be used for any binops.

Reply:
> Yes, that I understand, but why is the subclass even necessary as a subclass?

Not sure I understand what you mean, but we need the `ParameterArray` class so that we can mark the arrays that require gradients. Later on this won't be needed if we add a `gradient` field to dace arrays.